rusk-wallet/src/wallet.rs

```diff
@@ -529,31 +522,49 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
             return Err(Error::NotEnoughGas);
         }
 
+        let state = self.state()?;
+        let seed = self.store.get_seed();
+
         let mut rng = StdRng::from_entropy();
+        let amt = *amt;
         let sender_index = addr.index()?;
+        let sender_sk = derive_phoenix_sk(seed, sender_index);
+        let stake_sk = derive_bls_sk(seed, sender_index);
 
-        // stake
-        let tx = wallet.stake(
-            &mut rng,
-            sender_index as u64,
-            sender_index as u64,
-            addr.psk(),
-            *amt,
-            gas.limit,
-            gas.price,
+        let nonce = state
+            .fetch_stake(&AccountPublicKey::from(&stake_sk))?
+            .map(|s| s.nonce)
+            .unwrap_or(0);
+
+        let inputs = state
+            .inputs(sender_index, amt + gas.limit * gas.price)?
+            .into_iter()
+            .map(|(a, b, _)| (a, b))
+            .collect();
+
+        let root = state.fetch_root()?;
+        let chain_id = state.fetch_chain_id()?;
+
+        let stake = phoenix_stake::<_, Prover>(
+            &mut rng, &sender_sk, &stake_sk, inputs, root, gas.limit,
+            gas.price, chain_id, amt, nonce,
         )?;
-        Ok(tx)
+
+        state.prove_and_propagate(stake)
```

Comment:
here both stake and phoenix secret keys need to be zeroized

Reply:
Done